test/unit/proof/lrat_proof_black.h

```diff
@@ -0,0 +1,112 @@
+/*********************                                                        */
+/*! \file lrat_proof_black.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Alex Ozdemir
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Black box testing of the DRAT proof class
+ **
+ ** In particular, tests DRAT binary parsing.
+ **/
+
+#include <cxxtest/TestSuite.h>
+
+#include <cctype>
+#include <iostream>
+
+#include "proof/lrat/lrat_proof.h"
+#include "prop/sat_solver_types.h"
+
+using namespace CVC4::proof::lrat;
+using namespace CVC4::prop;
+
+class LfscProofBlack : public CxxTest::TestSuite
+{
+ public:
+  void setUp() override {}
+  void tearDown() override {}
+
+  void testOutputThreeAsLfsc();
+};
+
+void LfscProofBlack::testOutputThreeAsLfsc()
+{
+  std::vector<std::unique_ptr<LratInstruction>> instructions;
+
+  // 6   d 1 2
+  std::vector<ClauseIdx> clausesToDelete{1, 2};
+  std::unique_ptr<LratDeletion> deletion = std::unique_ptr<LratDeletion>(
+      new LratDeletion(6, std::move(clausesToDelete)));
+  instructions.push_back(std::move(deletion));
+
+  // 7   1 2 0  5 2 0
+  std::vector<SatLiteral> firstAddedClause{SatLiteral(1, false),
+                                           SatLiteral(2, false)};
+  LratUPTrace firstTrace{5, 2};
+  std::vector<std::pair<ClauseIdx, LratUPTrace>> firstHints;
+  std::unique_ptr<LratAddition> add1 =
+      std::unique_ptr<LratAddition>(new LratAddition(
+          7, std::move(firstAddedClause), std::move(firstTrace), firstHints));
+  instructions.push_back(std::move(add1));
+
+  // 8   2 0  -1 3  -5 2 0
+  std::vector<SatLiteral> secondAddedClause{SatLiteral(2, false)};
+  LratUPTrace secondTrace;
+  std::vector<std::pair<ClauseIdx, LratUPTrace>> secondHints;
+  LratUPTrace secondHints0Trace{3};
+  secondHints.emplace_back(1, secondHints0Trace);
+  LratUPTrace secondHints1Trace{2};
+  secondHints.emplace_back(5, secondHints1Trace);
+  std::unique_ptr<LratAddition> add2 = std::unique_ptr<LratAddition>(
+      new LratAddition(8,
+                       std::move(secondAddedClause),
+                       std::move(secondTrace),
+                       secondHints));
+  instructions.push_back(std::move(add2));
+
+  LratProof proof(std::move(instructions));
+
+
+  std::ostringstream lfsc;
+  proof.outputAsLfsc(lfsc);
+  std::ostringstream lfscWithoutWhitespace;
+  for (char c : lfsc.str())
+  {
+    if (!std::isspace(c))
+    {
+      lfscWithoutWhitespace << c;
+    }
+  }
```

Comment:
Suggestion: Create a helper function for this.

Reply:
Good idea! This is also an opportunity to use some STL.

Reply:
Nice :)